core/src/main/java/org/infinispan/notifications/impl/AbstractListenerImpl.java

```diff
@@ -143,7 +148,6 @@ public void removeListener(Object listener) {
    }
 
    private void removeListenerInvocation(Class<? extends Annotation> annotation, Object listener) {
-      if (listener == null) return;
```

Comment:
Listener cannot be null. It would fail earlier, here : https://github.com/anistor/infinispan/blob/t_indexed_dsl_filters/core/src/main/java/org/infinispan/notifications/impl/AbstractListenerImpl.java#L146